File: frontend/src/services/accountService.js
import apiClient from './apiClient.js';

export async function getAccounts() {
  const { data } = await apiClient.get('/accounts');
  return data.data;
}

export async function getAccount(id) {
  const { data } = await apiClient.get(`/accounts/${id}`);
  return data.data;
}

export async function createAccount(payload) {
  const { data } = await apiClient.post('/accounts', payload);
  return data.data;
}

export async function updateAccount(id, payload) {
  const { data } = await apiClient.put(`/accounts/${id}`, payload);
  return data.data;
}

export async function deleteAccount(id) {
  await apiClient.delete(`/accounts/${id}`);
}

export async function transfer(payload) {
  const { data } = await apiClient.post('/accounts/transfer', payload);
  return data.data;
}

File: backend/src/main/java/com/smartmoneymanager/backend/service/AccountService.java
package com.smartmoneymanager.backend.service;

import java.util.List;

import com.smartmoneymanager.backend.dto.request.CreateAccountRequest;
import com.smartmoneymanager.backend.dto.request.TransferRequest;
import com.smartmoneymanager.backend.dto.request.UpdateAccountRequest;
import com.smartmoneymanager.backend.dto.response.AccountResponse;
import com.smartmoneymanager.backend.dto.response.TransferResponse;

public interface AccountService {

    List<AccountResponse> getAccounts(Long userId);

    AccountResponse getAccount(Long userId, Long accountId);

    AccountResponse createAccount(Long userId, CreateAccountRequest request);

    AccountResponse updateAccount(Long userId, Long accountId, UpdateAccountRequest request);

    void deleteAccount(Long userId, Long accountId);

    TransferResponse transfer(Long userId, TransferRequest request);
}
